### ISSUE TITLE
Updating folder name

### ISSUE THREAD

**Issue by pschicho** [papis/papis#295]
Is there a way to "update" both the file- and folder name after an author name change without adding just the whole library entry again?

**Comment by alejandrogallo** [comment]
Hello @pschicho, yes there is, do you mean a general solution like in a script
or just for a single paper?

For a single paper you can do

```
papis rename
```

This will rename the folder you're using
interactively.

If you want a general solution as a script I can
give you an example of a python snippet for it and
put this question into the FAQ.

**Comment by pschicho** [comment]
Let me rephrase my question: Can the folder name and filename be changed/synchronised upon the information inside the info.yaml file? Since by adding the file through add-folder-name and add-file-name this gets fixed when adding.

**Comment by alejandrogallo** [comment]
Do you really want to rename the filename, I mean the `pdf` or files associated with your document?

What I can imagine that you want to do is
run a script through your library that changes the names of the folders
and the files according to the author and title of the current document.

I wrote this as an example of what you can accomplish
writing a small python script using the papis api.

```python
import papis.api
import papis.commands.rename
from papis.utils import clean_document_name

import shutil
import os

rename_files = True
library_name = "papers"

docs = papis.api.get_all_documents_in_lib(library_name)

for doc in docs:

    # get a new name formed by:
    #   -  10 characters of the author field
    #   -  10 characters of the title field
    #   -  the year, if any
    #
    new_name_unsafe = "{:.10}-{:.10}-{}".format(doc["title"],
                                                doc["author"],
                                                doc["year"])
    # This new name includes potentially spaces and weird characters
    # papis ships with a function to clean it up
    clean_name = clean_document_name(new_name_unsafe)

    print(clean_name)

    # This is the signature of the rename function
    #
    # def run(document: papis.document.Document,
    #          new_name: str, git: bool = False) -> None:
    #papis.commands.rename(doc, clean_name)

    files = doc.get_files()
    if rename_files and files:

        # TODO: GENERALIZE FOR MULTIPLE FILES AND DIFFERENT EXTENSIONS
        #       THAN PDFS
        original_file = files[0]

        if not files:
            continue

        new_name = os.path.join(os.path.dirname(original_file),
                                clean_name + ".pdf")
        print(original_file)
        print(new_name)

        shutil.move(original_file, new_name)
        doc["files"] = [new_name]
        doc.save()
```

As you see, you first go through the papers of your library,
create your new name, reuse the rename function from papis
and then you can manually move the associated files.

If you rename the associated files, then you have to set the `files`
field in the papis document and save the document like
`doc.save()`.

Feel free to accomodate this script to your needs.

Make sure to make a copy of your library before testing and modifying the
script ;)
